### PR TITLE
{bio}[foss/2019a] EnsEMBLCoreAPI v96.0-r20190601

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)
New easyconfig for EnsEMBL Core API. The version combines a release number and a date. The release number indicates the branch and relates to other EnsEMBL products (VEP). However, we have to target specific comits for each of its components because there are no sane releases of Core API. Hence, the date in the version.

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2e61d5a415e18bc9746ac585365fd98c for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/0d37ff6709179e5a9aeb8759737c9f3a for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/82afb3aab9a3073bbaaf78004e4a2a4f for a full test report.

File: easybuild/easyconfigs/e/EnsEMBLCoreAPI/EnsEMBLCoreAPI-96.0-r20190601-foss-2019a-Perl-5.28.1.eb [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        'source_tmpl': 'URI-1.76.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/O/OA/OALDERS'],
+        'checksums': ['b2c98e1d50d6f572483ee538a6f4ccc8d9185f91f0073fd8af7390898254413e'],
+    }),
+    ('XML::Writer', '0.625', {
+        'source_tmpl': 'XML-Writer-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/J/JO/JOSEPHW'],
+        'checksums': ['e080522c6ce050397af482665f3965a93c5d16f5e81d93f6e2fe98084ed15fbe'],
+    }),
+    ('XSLoader', '0.24', {
```
Comment (boegel) [inline easybuild/easyconfigs/e/EnsEMBLCoreAPI/EnsEMBLCoreAPI-96.0-r20190601-foss-2019a-Perl-5.28.1.eb]: @lexming Any reason not to include these extensions in the `Perl` easyconfig instead?

Comment (lexming) [reply easybuild/easyconfigs/e/EnsEMBLCoreAPI/EnsEMBLCoreAPI-96.0-r20190601-foss-2019a-Perl-5.28.1.eb]: @boegel the only reason is that these extensions are currently only used by this easyconfig afaik. But apart from this nothing hinders their inclusion in `Perl` instead. What do you prefer?

Comment (lexming) [reply easybuild/easyconfigs/e/EnsEMBLCoreAPI/EnsEMBLCoreAPI-96.0-r20190601-foss-2019a-Perl-5.28.1.eb]: @boegel revisiting this after more than 4 months. I moved `Set::InetrvalTree` and `URI::Escape` (and their ~40 dependencies) to main Perl easyconfig as these are used by other stuff (`STAR-Fusion`). The others are quite specific and some require non-Perl dependencies, so I'm leaving those in this easyconfig.

**Comment by lexming** [comment]
@migueldiascosta I'm updating this PR by moving `Set::IntervalTree` and `URI::Escape` (and their ~40 dependencies) to the main Perl easyconfig. These perl modules are also used in `STAR-Fusion`, so I'm removing them from `STAR-Fusion` as well. Apart from this, the installation of `STAR-Fusion` is equivalent to the existing one.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9a2af0595692ab969d17dfa05a4f3c96 for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
generoso-1 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/1a8df34b01108c0eb8a41bdfa6e3d8bb for a full test report.

File: easybuild/easyconfigs/p/Perl/Perl-5.28.1-GCCcore-8.2.0.eb [hunk 2]
```diff
@@ -1495,6 +1495,251 @@ exts_list = [
         'source_urls': ['https://cpan.metacpan.org/authors/id/J/JD/JDHEDDEN/'],
         'checksums': ['9dfd6ca2822724347e0eb6759d00709425814703ad5c66bdb6214579868bcac4'],
     }),
+    ('ExtUtils::MakeMaker', '7.36', {
```
Comment (boegel) [inline easybuild/easyconfigs/p/Perl/Perl-5.28.1-GCCcore-8.2.0.eb]: @lexming Do you mind fleshing this out in a separate PR, and adding the same extensions to `Perl-5.30.0-GCCcore-8.3.0.eb` as well?

Comment (lexming) [reply easybuild/easyconfigs/p/Perl/Perl-5.28.1-GCCcore-8.2.0.eb]: Sure, I've moved those changes to #10049

**Comment by lexming** [comment]
Depends on #10049

**Comment by lexming** [comment]
@boegel @migueldiascosta the required changes to Perl base install are already merged :+1:

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/5213df758584c57177032527c15bfcf9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in this PR)
node3100.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/644c30a9b6e2861f47bfe2023af6e613 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**